solutions/usaco-688.mdx

```diff
@@ -0,0 +1,60 @@
+---
+id: usaco-688
+source: USACO Bronze 2017 January
+title: Hoof, Paper, Scissors
+author: Ryan Chou
+---
+
+## Solution 1
+[Official Analysis](http://www.usaco.org/current/data/sol_hps_bronze_jan17.html)
+
+## Solution 2 - Generating All Permutations
+
+We can map the games to every possible permutation of the moves (hoof, paper, scissors) and find the sequence which gives us the maximum wins. This works because $1 \leq N \leq 100$.
+
+### Implementation
+
+**Time Complexity: ** $\mathcal{O}(n)$.
+<LanguageSection>
+<PySection> 
+
+```py
+from itertools import permutations
+import sys
+
+sys.stdin = open('hps.in', 'r')
+sys.stdout = open('hps.out', 'w')
+
+n = int(input())
+games = []
+
+for _ in range(n):
+	games.append(list(map(int, input().split())))
+
+# Original Ordering 
+# 1 = Hoof, 2 = Paper, 3 = Scissors
+choices = [1, 2, 3]
+
+# in these cases, cow 1 wins
+# hoof beats scissors, 
+# scissors beats paper, 
+# and paper beats hoof
```

Comment:
don't really think these comments are necessary? actually maybe it'd be better to just replace them with 
```py
# hoof -> scissors -> paper -> hoof
```

Reply:
thanks!